util/topgen/merge.py

```diff
@@ -892,29 +894,41 @@ def create_alert_lpgs(top, name_to_block: Dict[str, IpBlock]):
 
         # if clock group is "unique", add some uniquification to the tag
         lpg_name = f"{module['name']}_{lpg_name}" if unique_cg else lpg_name
-        if lpg_name not in lpg_dict:
-            lpg_dict.update({lpg_name: num_lpg})
+
+        def append_to_lpg_dict(lpg_dict):
             # since the alert handler can tolerate timing delays on LPG
             # indication signals, we can just use the clock / reset signals
             # of the first block that belongs to a new unique LPG.
             clock = module['clock_connections'][block_clock.clock]
-            top['alert_lpgs'].append({
+            lpg_dict.append({
                 'name': lpg_name,
                 'clock_group': None if unmanaged_clock else clock_group,
                 'clock_connection': clock,
                 'unmanaged_clock': unmanaged_clock,
                 'unmanaged_reset': is_unmanaged_reset(top, reset_name),
                 'reset_connection': primary_reset
             })
-            num_lpg += 1
+
+        alert_group = module.get('outgoing_alert')
+        if alert_group:
```

Comment:
Sorry to be annoying, but I'd probably suggest checking with `is not None` to avoid silly things happening if we accidentally have an alert group with an empty name.